src/Utilities/AnsiBuilder.cs

```diff
@@ -0,0 +1,330 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace WheelMUD.Utilities
+{
+    /// <summary>
+    /// Mutable String class for Ansi based strings, optimized for speed and memory while retrieving the final result
+    /// as a string. Similar use to StringBuilder, but avoid a lot of allocations done by StringBuilder
+    /// </summary>
+    public class AnsiBuilder
+    {
+        /// <summary>
+        ///     Working mutable string
+        /// </summary>
+        private char[] buffer;
+
+        private int bufferPos;
+        private int charsCapacity;
+
+        /// <summary>
+        /// Temporary string used for the Replace method
+        /// </summary>
+        private List<char> replacement;
+
+        /// <summary>
+        /// Immutable string. Generated at last moment, only if needed
+        /// </summary>
+        private string stringGenerated = "";
+
+        /// <summary>
+        /// Dirty flag for AnsiBuilder containing ansi that needs parsing
+        /// </summary>
+        private bool needsParsing;
+
+        public AnsiBuilder(int initialCapacity = 32)
+        {
+            buffer = new char[charsCapacity = initialCapacity];
```

Comment:
StringBuilder is better when you are appending extreme amounts of strings. But with MUDS you shouldn't be appending more than a few lines at a time, the AnsiBuilder is created with that in mind.

AnsiBuilder also has no unused features, this allows for cleaner code being written from developers, and a more uniform approach (since there are tons of ways to accomplish the same thing with StringBuilder)

Id be happy to provide some tests to show the performance improvements over using StingBuilder. 

The AnsiBuilder allows you to adjust the size of the initial buffer size. So if you know the initial size, and the message isn't being changed, then you can pass in the message length before appending. Or if you know its going to be really small make it smaller before appending, etc.

Reply:
Here is a repo with a basic comparison between the two - [https://github.com/DrunkenImpGameStudio/AnsiBuilder](url)

basic results (mean of 1000 iterations);

Test result: StringBuilder
normalized mean: 4841.052775

Test result: AnsiBuilder
normalized mean: 60.14525

Reply:
Fantastic!  Thanks for following up on this.